### ISSUE TITLE
Crash during reports refresh

### ISSUE THREAD

**Issue by Jean-BaptisteC** [Exodus-Privacy/exodus-android-app#110]
Issue:

- Device: Pixel 6
- OS: Android 12
- EP Version: Preview

How reproduce:

- refresh reports
- disable Internet during refresh
- App crashed

**Comment by Jean-BaptisteC** [comment]
| Version | Detected |
| --- | --- |
| Android 5 | Yes |
| Android 7 | No|
| Android 8 | No |
| Android 12 | Yes |

**Comment by Jean-BaptisteC** [comment]
* Stacktrace on Pixel 6 - Android 12
```
04-09 14:52:35.857  2340  2548 E AndroidRuntime: FATAL EXCEPTION: Thread-51
04-09 14:52:35.857  2340  2548 E AndroidRuntime: Process: org.eu.exodus_privacy.exodusprivacy, PID: 2340
04-09 14:52:35.857  2340  2548 E AndroidRuntime: DeadSystemException: The system died; earlier logs will point to the root cause
04-09 14:52:35.863  2340  2551 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.864  2340  2524 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.875  2340  2576 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.879  2340  2560 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.888  2340  2512 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.888  2340  2578 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.889  2340  2562 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.889  2340  2582 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.899  2340  2586 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
04-09 14:52:35.902  2340  2580 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 96)
04-09 14:52:35.962  1525  3295 I ActivityManager: Process org.eu.exodus_privacy.exodusprivacy (pid 2340) has died: fg  TOP
```

**Comment by Jean-BaptisteC** [comment]
Pixel 6 - Android 12

```
05-23 15:19:21.226  1521  1684 E WifiScanRequestProxy: Failed to retrieve wifiscanner
05-23 15:19:21.227  1521  2663 E WifiService: Failed to start scan
05-23 15:19:21.230  1521  1560 I ActivityManager: Start proc 25747:com.android.carrierdefaultapp/u0a79 for broadcast {com.android.carrierdefaultapp/com.android.carrierdefaultapp.CarrierDefaultBroadcastReceiver}
05-23 15:19:21.251  1521  2663 E WifiService: Attempt to retrieve passpoint with invalid scanResult List
05-23 15:19:21.258 25747 25747 E rrierdefaultap: Not starting debugger since process cannot load the jdwp agent.
05-23 15:19:21.270  2236 25265 E WakeLock: GCM_HB_ALARM release without a matched acquire!
05-23 15:19:21.271 25054 25084 E AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-1
05-23 15:19:21.271 25054 25084 E AndroidRuntime: Process: org.eu.exodus_privacy.exodusprivacy, PID: 25054
05-23 15:19:21.275  2236 25265 E WakeLock: GCM_CONN_ALARM release without a matched acquire!
05-23 15:19:21.350 24109 24109 E cr_BTSPrefs: No data found for task id: 53
05-23 15:19:21.351 24109 24109 E cr_BkgrdTaskScheduler: Task cannot be canceled because no data was found instorage or data was invalid
05-23 15:19:21.416  1521  8359 I ActivityManager: Process org.eu.exodus_privacy.exodusprivacy (pid 25054) has died: fg +50 FGS
05-23 15:19:21.417  1521  8359 W ActivityManager: Scheduling restart of crashed service org.eu.exodus_privacy.exodusprivacy/.ExodusUpdateService in 50128ms for start-requested
```

**Comment by Jean-BaptisteC** [comment]
Fixed in 3.0
